modules/presence/subscriber.c

```diff
@@ -280,6 +280,45 @@ static int presence_alloc(struct contact *contact)
 	return 0;
 }
 
+static void presence_dealloc(struct presence* pres)
+{
+	list_unlink(&pres->le);
+	tmr_cancel(&pres->tmr);
+	mem_deref(pres->contact);
+}
```

Comment:
this looks a bit suspicious. the pointer to `pres->contact` must be set to `NULL`
after `mem_deref`.

perhaps it is better to deref the whole `struct presence` object ?